---
title: Puppy growth report
author: The mdhtml project
---

# Puppy growth report

This notebook is a small `viewmd` demo: a plain `.ipynb` rendered as a page.
Code cells keep their stored outputs^[Outputs are shown in an `output` block
under each code cell, with a dim orange border.], and everything else is
ordinary Markdown - cross-references, footnotes, and math included.

## Growth data {#sec-data}

We weighed the puppy[^who] three times. The measurements are in
[@tbl-weights], the subject is pictured in [@fig-puppy], and the analysis
lives in [@sec-math].

[^who]: A border collie, born early 2026.

In [ ]:
weights = {1: 2.1, 4: 3.4, 8: 5.0}
for week, kg in weights.items(): print(f"week {week}: {kg} kg")

week 1: 2.1 kg
week 4: 3.4 kg
week 8: 5.0 kg


| Week | Weight (kg) |
|-----:|------------:|
|    1 |         2.1 |
|    4 |         3.4 |
|    8 |         5.0 |
: Puppy weights by week {#tbl-weights}

![A very good dog, week eight](puppy.jpg){#fig-puppy width="240"}

## A little analysis {#sec-math}

Average weight is the usual arithmetic mean:

\[
\bar{w} = \frac{1}{n}\sum_{i=1}^{n} w_i
\]

For the data in [-@tbl-weights], \( n = 3 \):

In [ ]:
avg = sum(weights.values()) / len(weights)
round(avg, 2)

3.5

So the puppy of [@fig-puppy] averaged \( \bar{w} = 3.5 \) kg over the
period covered by [@tbl-weights] - respectable growth for [@sec-data]'s
eight weeks.